src/resources/ble_esp32.cc

```diff
@@ -759,17 +794,43 @@ uint32_t BleResourceGroup::on_event(Resource* resource, word data, uint32_t stat
   return state;
 }
 
+BleDescriptorResource::~BleDescriptorResource() {
+  characteristic_->remove_descriptor(this);
+}
+
 BleServiceResource* BleDescriptorResource::service() {
   return characteristic_->service();
 }
 
+BleCharacteristicResource::~BleCharacteristicResource() {
+  while (!subscriptions_.is_empty()) {
+    auto subscription = subscriptions_.remove_first();
+    delete subscription;
+  }
+  while (!descriptors_.is_empty()) {
+    auto descriptor = descriptors_.first();
+    group()->unregister_resource(descriptor);
```

Comment:
Maybe add a comment that explains that unregistering will remove the element from linked list?

Reply:
done.